waspc/src/Wasp/Analyzer/Parser/Ctx.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE DeriveFunctor #-}
```

Comment:
This is where `Ctx` and `WithCtx` is defined. This parsing context is then used through the whole Analyzer codebase, to keep track of information needed mostly to report errors. We keep it in ASTs, and we propagate it to errors.